### PR TITLE
net-news/newsboat: add 2.34, ship prebuilt docs

### PR DESCRIPTION
**Pull request by zlamas**
Closes: https://bugs.gentoo.org/831123
Closes: https://bugs.gentoo.org/901213
Closes: https://bugs.gentoo.org/910210

### CODE REVIEW

**Review by antecrescent — CHANGES_REQUESTED** [review]
See header [The variables ED and EROOT](https://wiki.gentoo.org/wiki/Project:Prefix/Technical_Documentation).

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	virtual/pkgconfig
+	>=virtual/rust-1.70.0
+"
+
+src_prepare() {
+	default
+
+	sed -i \
+		-e "s/WARNFLAGS=-Werror -Wall/WARNFLAGS=-Wall/" \
+		-e "s/BARE_CXXFLAGS=-std=c++11 -O2 -ggdb/BARE_CXXFLAGS=-std=c++11/" \
```
Comment (antecrescent) [inline net-news/newsboat/newsboat-2.34.ebuild:164]: You could patch the `doc` target and ship the prebuilt manpages instead as per https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0305.
i.e.
`sed -i "s/^doc: .*\.1\(.*\)/doc: \1/" Makefile` here and `doman "${FILESDIR}"/*.1` in `src_install`.

**Comment by zlamas** [comment]
> Is there any particular reason, why you opened a new PR instead of bumping the old one? I'm just curious :)

I wasn't sure what to do, I thought I had to open another one for a new version. Should I close the old one?

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+src_configure() {
+	if tc-is-gcc; then
+		filter-lto  # bug #877657
+	fi
+	./config.sh || die
+}
+
+src_compile() {
+	export CARGO_HOME="${ECARGO_HOME}"
+	emake prefix="/usr" CXX="$(tc-getCXX)" AR="$(tc-getAR)" RANLIB="$(tc-getRANLIB)" \
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: Done

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+		newsboat podboat mo-files $(usev doc)
+}
+
+src_test() {
+	emake CXX="$(tc-getCXX)" AR="$(tc-getAR)" RANLIB="$(tc-getRANLIB)" check
+}
+
+src_install() {
+	emake DESTDIR="${D}" prefix="/usr" docdir="/usr/share/doc/${PF}" \
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: Done

File: net-news/newsboat/newsboat-2.34.ebuild (hunk 1 above)
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:164]: In this case, where should I put the manpages?

Comment (antecrescent) [reply net-news/newsboat/newsboat-2.34.ebuild:164]: Put any of the manpages <=20KiB into a subdirectory called [`files`](https://devmanual.gentoo.org/ebuild-maintenance/new-ebuild/#the-files-directory). You can then access it via the `FILESDIR` variable. Larger files should be hosted elsewhere and downloaded together with the other source files.

**Comment by antecrescent** [comment]
> > Is there any particular reason, why you opened a new PR instead of bumping the old one? I'm just curious :)
> 
> I wasn't sure what to do, I thought I had to open another one for a new version. Should I close the old one?
Usually, you just keep pushing onto the original branch (update the PR title) and occasionally rebase onto master. There's no need to close the old one. The dev who merges this commit can close both PRs in one go.

File: net-news/newsboat/newsboat-2.34.ebuild (hunk 1 above)
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:164]: Can I host them on my github?

Comment (antecrescent) [reply net-news/newsboat/newsboat-2.34.ebuild:164]: Sure.

Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:162]: Done

**Comment by zlamas** [comment]
For some reason, `install-docs` make target also installs user contributed themes and scripts, so I decided to drop the use flag and always install docs. Also cleaned up the ebuild.

**Comment by zlamas** [comment]
> > For some reason, `install-docs` make target also installs user contributed themes and scripts, so I decided to drop the use flag and always install docs. Also cleaned up the ebuild.
> 
> My bad, I should have caught that before giving advice. I'm not sure if installing html documentation unconditionally is the correct way to do this. Let me ask on IRC and get back to you before committing any changes.

Any news on how we should proceed?

**Review by antecrescent — CHANGES_REQUESTED** [review]
Sorry for the late reply. I took me quite a lot more time than expected to review this ebuild. Shipping prebuilt manpages and the HTML docs is fine since the latter are only ~370KiB in size.

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	if tc-is-gcc ; then
+		filter-lto  # bug #877657
+	fi
+}
+
+src_compile() {
+	default
+}
+
+src_test() {
```
Comment (antecrescent) [inline net-news/newsboat/newsboat-2.34.ebuild:182]: Please add `export TMPDIR="${T}"`

**Review by antecrescent — CHANGES_REQUESTED** [review]

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+	>=virtual/rust-1.70.0
+"
+
+src_prepare() {
+	default
+
+	sed -i \
+		-e "s/WARNFLAGS=-Werror -Wall/WARNFLAGS=-Wall/" \
+		-e "s/BARE_CXXFLAGS=-std=c++11 -O2 -ggdb/BARE_CXXFLAGS=-std=c++11/" \
+		-e "s/^doc: .*/doc:/" \
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: Done

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+DESCRIPTION="An RSS/Atom feed reader for text terminals"
+HOMEPAGE="https://newsboat.org/ https://github.com/newsboat/newsboat"
+SRC_URI="
+	https://newsboat.org/releases/${PV}/${P}.tar.xz
+	https://github.com/zlamas/newsboat-docs/archive/refs/tags/${PV}.tar.gz -> ${P}-docs.tar.gz
+	${CARGO_CRATE_URIS}
+"
+
+LICENSE="Apache-2.0 Apache-2.0-with-LLVM-exceptions Boost-1.0 BSD MIT Unlicense ZLIB"
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: Correction: logo's license is `CC-BY-4.0`.
newsboat itself is only MIT, moved the rest to crate licenses.

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+	local docdir="${WORKDIR}/${PN}-docs-${PV}"
+	mkdir doc/xhtml || die
+	mv "${docdir}"/*.1 doc || die
+	mv "${docdir}"/*.html doc/xhtml || die
+}
+
+src_configure() {
+	if tc-is-gcc ; then
+		filter-lto  # bug #877657
+	fi
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:193]: Done

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+	sys-libs/zlib
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+"
+BDEPEND="
+	app-alternatives/awk
+	virtual/pkgconfig
+	>=virtual/rust-1.70.0
+"
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: Done

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+	sed -i \
+		-e "s/WARNFLAGS=-Werror -Wall/WARNFLAGS=-Wall/" \
+		-e "s/BARE_CXXFLAGS=-std=c++11 -O2 -ggdb/BARE_CXXFLAGS=-std=c++11/" \
+		-e "s/^doc: .*/doc:/" \
+		Makefile || die
+
+	local docdir="${WORKDIR}/${PN}-docs-${PV}"
+	mkdir doc/xhtml || die
+	mv "${docdir}"/*.1 doc || die
+	mv "${docdir}"/*.html doc/xhtml || die
```
[the remark this answers is not in the document]
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:186]: Done

File: net-news/newsboat/newsboat-2.34.ebuild (hunk 4 above)
Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:205]: Done

File: net-news/newsboat/newsboat-2.34.ebuild [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+HOMEPAGE="https://newsboat.org/ https://github.com/newsboat/newsboat"
+SRC_URI="
+	https://newsboat.org/releases/${PV}/${P}.tar.xz
+	https://github.com/zlamas/newsboat-docs/archive/refs/tags/${PV}.tar.gz -> ${P}-docs.tar.gz
+	${CARGO_CRATE_URIS}
+"
+
+LICENSE="MIT CC-BY-4.0"
+# Dependent crate licenses
+LICENSE+=" Apache-2.0 Boost-1.0 Unicode-DFS-2016"
```
Comment (antecrescent) [inline net-news/newsboat/newsboat-2.34.ebuild:136]: Please move the bundled third-party code licenses `Apache-2.0` and `Boost-1.0` back up and sort the `LICENSE` expression. We don't want to risk confusing them with crate licenses and deleting them upon a version bump.
And thank you for catching my typo!

Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild:136]: Oh, sorry, I didn't consider that. Though I don't see any `Apache-2.0` licensed bundled code, am I missing something?

Comment (antecrescent) [reply net-news/newsboat/newsboat-2.34.ebuild:136]: `grep -rl Apache 3rd-party` reveals that `3rd-party/json.hpp` contains code licensed under `Apache-2.0`. :)

Comment (zlamas) [reply net-news/newsboat/newsboat-2.34.ebuild]: I see now, thank you. Fixed

**Comment by zlamas** [comment]
What's the next step?

**Comment by antecrescent** [comment]
> What's the next step?

Wait for a dev who is interested to review and merge this. If there's no activity for ~a week you could ask on IRC for help with getting this merged.

**Comment by antecrescent** [comment]
@cedk ping :)

**Review by arthurzam — APPROVED** [review]

**Pull request closed without merging**